cylc/flow/task_pool.py

```diff
@@ -1244,8 +1269,13 @@ def spawn_on_all_outputs(self, itask):
                     continue
                 # Spawn child only if itask.flow_nums is not empty.
                 c_task = self.spawn_task(c_name, c_point, itask.flow_nums)
-                if c_task is not None:
-                    self.add_to_pool(c_task)
+                if completed_only:
+                    c_task.state.satisfy_me({
+                        (str(itask.point), itask.tdef.name, output)
+                    })
+                    self.data_store_mgr.delta_task_prerequisite(c_task)
```

Comment:
This branch isn't covered, need to manually test...

Reply:
I've run the test from #4651 against a flow.cylc and a suite.rc variants and god the same result, happy.

Reply:
Something wrong with code coverage here.  The functional test on this branch should go into this branch...

... brute force test: I put a `print` statement in there and ran the test, it does get executed. :thinking:

Reply:
Any idea what could cause coverage to miss this?

Reply:
I think the coverage is accurate in this case. I put a `LOG.critical("Here")` at L1273 and it didn't show up in the workflow log

Reply:
Doesn't this branch only get hit in Cylc 7 back-compat mode?

I tested back-compat to make sure it worked, but didn't run coverage or insert a marker to test whether it hit this branch.

Reply:
No, `completed_only` is True for retroactive spawning on completed outputs (when an ongoing flow merges with a one-off manually-triggered task (thus making it part of the ongoing flow) ... i.e. what this PR is about). In back-compat mode, `completed_only` is False and we spawn on all outputs ahead of time (before they are completed) to approximate Cylc 7 spawning of tasks before they are needed.

Reply:
> I think the coverage is accurate in this case. I put a `LOG.critical("Here")` at L1273 and it didn't show up in the workflow log for `tests/functional/spawn-on-demand/14-trigger-flow-blocker.t`

Strange. I just tried the same, to double check, and it does show up in my workflow log. (I did have to disable the purge in the `reftest` shell function, to keep the log for inspection).

Reply:
(Maybe the test is not actually working correctly in some environments, namely yours and GitHub actions ... I'll investigate later...)

Reply:
Test fixed. It was flaky.